server/routerlicious/packages/lambdas/src/scriptorium/lambda.ts

```diff
@@ -88,12 +97,15 @@ export class ScriptoriumLambda implements IPartitionLambda {
         Promise.all(allProcessed).then(
             () => {
                 this.current.clear();
+                metric?.setProperty("processingComplete", true);
                 // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                 this.context.checkpoint(batchOffset!);
+                metric?.setProperty("checkpointComplete", true);
```

Comment:
Instead of keeping two properties, maybe we can consider a property of `state`, and set it to following like:
running,
saved to DB
Processing completed
checkpoint completed

Reply:
Thats a good idea. Will update, thanks :)